src/vaadin-text-field-mixin.html

```diff
@@ -424,7 +424,8 @@
         }
       }
 
-      if (!e.__fromClearButton) {
+      // Value can remain the same on input when pasting
+      if (!e.__fromClearButton && this.value !== e.target.value) {
```

Comment:
Alternative fix would be to set `this.__userInput` back to `false` after `this.value` is set on line 430. Then we wouldn't have to rely on some side effect code somewhere to fix the flag back to `false`. wdyt?

Reply:
Currently there is no need in setting the flag back to `false` as it won't be set to `true` incorrectly in the first place.
So I am avoiding one more place of removing the `userInput` flag this way.

But I don't mind other solution if it seems to be cleaner and I can use that one if needed :)

Reply:
> So I am avoiding one more place of removing the userInput flag this way

The place where it's currently set `false` could be removed, so it wouldn't add new logic but streamlines the existing logic.

The current removal of the flag is done somewhere else that relies on a side-effect of another action to trigger and that caused the issue in the first place (the observer didn't invoke if the set value happened to be the same as before).

IMO, toggling a flag like this would ideally be done in only one place. Now the logic is scattered so it's hard to follow and subject to errors.

Reply:
Thanks for the clarification! That makes sense indeed as scattered logic was the cause of the issue I guess 😅 . 
Didn't consider moving removal from `_valueChanged`. Updated now.